java/lang/Float/FloatTest.cpp

```diff
@@ -441,124 +449,1464 @@ TEST(JavaLang, FloatShortValue) {
+    ASSERT_STR(expectedResultToString , actualResultToString);
+
+    // Test MIN_NORMAL
+    variableTestToString = MIN_NORMAL;
+    expectedResultToString = (string) "1.4013e-45";
+    actualResultToString = variableTestToString.toString();
+    ASSERT_STR(expectedResultToString , actualResultToString);
+
+    // Test MIN_VALUE
+    variableTestToString = MIN_VALUE;
+    expectedResultToString = (string) "1.17549e-38";
+    actualResultToString = variableTestToString.toString();
+    ASSERT_STR(expectedResultToString , actualResultToString);
+}
+
+TEST(JavaLang, FloatParseFloat) {
+    // Test POSITIVE_INFINITY
+    ASSERT_FLOAT_NEAR(POSITIVE_INFINITY, Float::parseFloat((String) "inf").floatValue());
+
+    // Test NEGATIVE_INFINITY
+    ASSERT_FLOAT_NEAR(NEGATIVE_INFINITY, Float::parseFloat((String) "-inf").floatValue());
+
+    // Test NaN_NUMBER
+    ASSERT_FLOAT_NEAR(-NaN_NUMBER, Float::parseFloat((String) "-nan").floatValue());
+
+    // Test MAX_VALUE
+    ASSERT_FLOAT_NEAR(MAX_VALUE,
+                      Float::parseFloat((String) "340282346638528859811704183484516925440.0").floatValue());
+
+    // Test MIN_NORMAL
+    ASSERT_FLOAT_NEAR(MIN_NORMAL, Float::parseFloat((String) "1.4013e-45").floatValue());
+
+    // Test MIN_VALUE
+    ASSERT_FLOAT_NEAR(MIN_VALUE, Float::parseFloat((String) "1.17549e-38").floatValue());
+}
+
+TEST(JavaLang, FloatValueOf) {
+    // Test POSITIVE_INFINITY
+    ASSERT_FLOAT_NEAR(POSITIVE_INFINITY, Float::valueOf((String) "inf").floatValue());
+
+    // Test NEGATIVE_INFINITY
+    ASSERT_FLOAT_NEAR(NEGATIVE_INFINITY, Float::valueOf((String) "-inf").floatValue());
+
+    // Test NaN_NUMBER
+    ASSERT_FLOAT_NEAR(-NaN_NUMBER, Float::valueOf((String) "-nan").floatValue());
+
+    // Test MAX_VALUE
+    ASSERT_FLOAT_NEAR(MAX_VALUE,
+                      Float::valueOf((String) "340282346638528859811704183484516925440.0").floatValue());
+
+    // Test MIN_NORMAL
+    ASSERT_FLOAT_NEAR(MIN_NORMAL, Float::valueOf((String) "1.4013e-45").floatValue());
+
+    // Test MIN_VALUE
+    ASSERT_FLOAT_NEAR(MIN_VALUE, Float::valueOf((String) "1.17549e-38").floatValue());
+}
+
+TEST(JavaLang, FloatValueOf2) {
+    // Test POSITIVE_INFINITY
+    ASSERT_FLOAT_NEAR(POSITIVE_INFINITY, Float::valueOf(POSITIVE_INFINITY).floatValue());
+
+    // Test NEGATIVE_INFINITY
+    ASSERT_FLOAT_NEAR(NEGATIVE_INFINITY, Float::valueOf(NEGATIVE_INFINITY).floatValue());
+
+    // Test NaN_NUMBER
+    ASSERT_FLOAT_NEAR(-NaN_NUMBER, Float::valueOf(-NaN_NUMBER).floatValue());
+
+    // Test MAX_VALUE
+    ASSERT_FLOAT_NEAR(MAX_VALUE,
+                      Float::valueOf(340282346638528859811704183484516925440.0f).floatValue());
+
+    // Test MIN_NORMAL
+    ASSERT_FLOAT_NEAR(MIN_NORMAL, Float::valueOf(1.4013e-45).floatValue());
+
+    // Test MIN_VALUE
+    ASSERT_FLOAT_NEAR(MIN_VALUE, Float::valueOf(1.17549e-38).floatValue());
+}
+
+// TODO(thoangminh): need to pass test case in comment block
```

Comment:
Please enable tests

Reply:
Thank. I think I will be back to improve this method later to pass these tests. 
This method can't pass them now.

Reply:
What do you need to be pass?
I'll support as many as possible. :heart:

Reply:
Cool. The problem is that I need to improve the method Float::floatToBinary32StringType(float floatInput).

Reply:
You can improve it, just need to run the same as in this link: http://binary-system.base-conversion.ro/convert-real-numbers-from-decimal-system-to-32bit-single-precision-IEEE754-binary-floating-point.php